config/src/main/java/com/typesafe/config/impl/SimpleConfig.java

```diff
@@ -248,6 +248,19 @@ public String getString(String path) {
     }
 
     @Override
+    public <T extends Enum<T>> T getEnum(Class<T> enumClass, String path) {
+        ConfigValue v = find(path, ConfigValueType.STRING);
+        String enumName = (String) v.unwrapped();
+        try {
+            return Enum.valueOf(enumClass, enumName);
+        } catch (IllegalArgumentException e) {
+            throw new ConfigException.BadValue(
+                v.origin(), path,
+                "The enum class %s has no constant of the name %s".format(enumClass.getSimpleName(), enumName));
```

Comment:
Good idea, done. And, added an assertion to verify that the error message contains the list of valid `Enum` constants.